pandas/plotting/_core.py

```diff
@@ -573,6 +573,8 @@ def _get_xticks(self, convert_period=False):
                 self.data = self.data[notna(self.data.index)]
                 self.data = self.data.sort_index()
                 x = self.data.index._mpl_repr()
+            elif all(isinstance(val, string_types) for val in index):
```

Comment:
This will be slow for large indexes. Try `index.is_object()`.

Reply:
Is `is_object` equal to `is_str`? The reason for the generator is that in my example the datatype of the index was wrong (i.e. object and not str) and the generator is only slow when there are str objects. My assumption was that in the case that the index contains str, the index size is relative small.

Reply:
you can do
``index.inferred_type in ['string', 'unicode']`` is pretty performant

Reply:
ok, I changed the code to use `index.inferred_type`